image-sync/oc-mirror/deploy/image-sync-template.yml

```diff
@@ -0,0 +1,149 @@
+---
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: image-sync-template
+
+parameters:
+  - name: NAMESPACE
+    required: true
+    value: aro-hcp
+  - name: OC_MIRROR_IMAGE
+    required: true
+  - name: OC_MIRROR_CLIENT_ID
+    required: true
+    description: "Client ID of Frontend Managed Identity"
+  - name: TENANT_ID
+    description: "Tenant ID of the Subscription"
+    required: true
+  - name: KEYVAULT_NAME
+    description: "Name of the KeyVault service in Azure"
+    required: true
+  - name: TARGET_REGISTRY
+    description: "Name of the target registry"
+    required: true
+  - name: PULL_SECRET_NAME
+    description: "Name of the pull secret in the KeyVault"
+    required: true
+
+objects:
+  - apiVersion: v1
+    kind: Namespace
+    metadata:
+      name: ${NAMESPACE}
+  - apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      annotations:
+        azure.workload.identity/client-id: "${OC_MIRROR_CLIENT_ID}"
+      name: image-sync
+      namespace: ${NAMESPACE}
+  - apiVersion: batch/v1
+    kind: CronJob
+    metadata:
+      labels:
+        run: ocmirror
+      name: ocmirror
+      namespace: ${NAMESPACE}
+    spec:
+      schedule: "0 * * * *"
+      jobTemplate:
+        spec:
+          template:
+            metadata:
+              labels:
+                run: ocmirror
+                azure.workload.identity/use: "true"
+            spec:
+              initContainers:
+              - name: login
+                image: mcr.microsoft.com/azure-cli:cbl-mariner2.0
+                command: ['sh', '-c',
+                          "az login --federated-token $(cat $AZURE_FEDERATED_TOKEN_FILE) --service-principal -u $AZURE_CLIENT_ID -t $AZURE_TENANT_ID; accessToken=$(az acr login --name ${TARGET_REGISTRY} --expose-token | grep accessToken |cut -d ':' -f2| tr -d ' \",') ; cat /tmp/secret-orig/pull-secret | base64 -d  |sed \"s/TOKENTOBEREPLACED/$accessToken/\" > /etc/containers/auth.json",
```

Comment:
I think when we've done this in the past the token here is only good for an hour.  It shouldn't be a problem because we're running on Azure and will have higher bandwidth + we'll also have the ability to rerun it and hour later, but just a heads-up.

Reply:
Why do we need to use the secrets store driver if the managed identity the oc mirror pod has push permissions into ACR?

Reply:
> Why do we need to use the secrets store driver if the managed identity the oc mirror pod has push permissions into ACR?  

For the quay.io and registry.redhat.io pull secrets